lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -411,6 +411,7 @@ public function __construct($initializer = null, $cloner = null)
      * @param \Doctrine\Common\Persistence\Mapping\ClassMetadata $class
      *
      * @return string
+     * @throws \ReflectionException
```

Comment:
Please remove, besides being private API only, this can never happen as we check for `hasMethod()` before calling `getMethod()`.